04-vue-cli/03-UiDropdown1/components/UiDropdown.vue

```diff
@@ -23,8 +37,53 @@ import UiIcon from './UiIcon.vue';
 
 export default {
   name: 'UiDropdown',
-
   components: { UiIcon },
+  props: {
+    options: {
+      type: Array,
+      required: true,
+    },
+    modelValue: {
+      required: true,
+    },
+    title: {
+      type: String,
+      required: true,
+    }
+  },
+  data() {
+    return {
+      isOpen: false,
+      selectedItem: null,
+    };
+  },
+  emits: ['update:modelValue'],
+  computed: {
+    hasIcon() {
+      // eslint-disable-next-line no-prototype-builtins
+      return this.options.some((option) => option.hasOwnProperty('icon'))
+    }
```

Comment:
Falsy значение `icon` всё равно не может считаться иконкой, можно проверять на истинность поля:

```suggestion
    hasIcon() {
      return this.options.some((option) => option.icon)
    }
```